segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/TableBasedMetadataStore.java

```diff
@@ -167,29 +190,36 @@ private StorageMetadataException handleException(Throwable e) throws StorageMeta
         return new StorageMetadataException("Transaction failed", e);
     }
 
-    private void ensureInitialized() {
+    private CompletableFuture<Void> ensureInitialized() {
         if (!isTableInitialized.get()) {
             // Storage Metadata Segment is a System, Internal Segment. It must also be designated as Critical since the
             // Segment Store may not function properly without it performing well. The Critical designation will cause
             // all of its "modify" operations to bypass any ingestion pipeline throttling and be expedited for processing.
             val segmentType = SegmentType.builder().tableSegment().system().critical().internal().build();
-            try {
-                this.tableStore.createSegment(tableName, segmentType, timeout).join();
-                log.info("Created table segment {}", tableName);
-            } catch (CompletionException e) {
-                if (e.getCause() instanceof StreamSegmentExistsException) {
-                    log.info("Table segment {} already exists.", tableName);
-                }
-            }
-            isTableInitialized.set(true);
+            return this.tableStore.createSegment(tableName, segmentType, timeout)
+                    .thenRunAsync(() -> {
+                        log.info("Created table segment {}", tableName);
+                        isTableInitialized.set(true);
+                    }, getExecutor())
+                    .exceptionally(e -> {
+                        val ex = Exceptions.unwrap(e);
+                        if (e.getCause() instanceof StreamSegmentExistsException) {
+                            log.info("Table segment {} already exists.", tableName);
```

Comment:
debug. I don't need to see this upon every container boot. 

The one above where you create is OK to be info. It should only happen once.

Reply:
done.